tephra-hbase-compat-0.98/src/main/java/co/cask/tephra/hbase98/coprocessor/TransactionCellSkipFilter.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.tephra.hbase98.coprocessor;
+
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.KeyValue;
+import org.apache.hadoop.hbase.filter.Filter;
+import org.apache.hadoop.hbase.filter.FilterBase;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * {@link Filter} that uses {@link TransactionVisibilityFilter} internally. However it also
+ * remembers the last {@link Cell} for which it returned the {@link ReturnCode#NEXT_COL}, so that when
+ * {@link #filterKeyValue} is called again for the same {@link Cell} with different version, it returns
+ * {@link ReturnCode#NEXT_COL} directly without consulting the {@link TransactionVisibilityFilter}.
+ * Please see TEPHRA-169 for more details.
+ */
+public class TransactionCellSkipFilter extends FilterBase {
```

Comment:
Since this does not have any special logic for transaction, what do you say about just calling it `CellSkipFilter`?